File: Week2 Arrays Problems/Readability.c
#include <cs50.h>
#include <ctype.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    string sen = get_string("enter sentence: ");

    int letters = 0;
    for(int i = 0; i < strlen(sen); i++)
    {
        if(isalpha(sen[i]))
        {
            letters++;
        }
    }

    int words = 0;
    for(int i = 0; i < strlen(sen); i++)
    {
        if(sen[i] == ' ')
        {
            words++;
        }

    }
    words++;


    int tsen = 0;
     for(int i = 0; i < strlen(sen); i++)
    {
        if(sen[i] == '!' || sen[i] == '.' || sen[i] == '?')
        {
            tsen++;
        }
    }

   float l = ((float)letters/words) * 100;
   float s = ((float)tsen/words) * 100;

   float grade = round(0.0588 * l - 0.296 * s - 15.8);
   int index = (int) grade;

   if(index < 1)
   {
    printf("Before Grade 1\n");
   }
   else if(index == 1)
   {
    printf("Grade 1\n");
   }
   else if(index == 2)
   {
    printf("Grade 2\n");
   }
   else if(index == 3)
   {
    printf("Grade 3\n");
   }
   else if(index == 4)
   {
    printf("Grade 4\n");
   }
   else if(index == 5)
   {
    printf("Grade 5\n");
   }
   else if(index == 6)
   {
    printf("Grade 6\n");
   }
   else if(index == 7)
   {
    printf("Grade 7\n");
   }
   else if(index == 8)
   {
    printf("Grade 8\n");
   }
   else if(index == 9)
   {
    printf("Grade 9\n");
   }
   else if(index == 10)
   {
    printf("Grade 10\n");
   }
   else if(index == 11)
   {
    printf("Grade 11\n");
   }
   else if(index == 12)
   {
    printf("Grade 12\n");
   }
   else if(index == 13)
   {
    printf("Grade 13\n");
   }
   else if(index == 14)
   {
    printf("Grade 14\n");
   }
   else if(index == 15)
   {
    printf("Grade 15\n");
   }
   else if(index == 16)
   {
    printf("Grade 16\n");
   }
   else
   {
    printf("Grade 16+\n");
   }

}

File: Week6 Python Problems/readability.py
# readability.py
from cs50 import get_string

def count_letters(text):
    return sum(1 for c in text if c.isalpha())

def count_words(text):
    # Words are separated by spaces
    return len(text.split())

def count_sentences(text):
    # Sentences end with ., !, or ?
    return sum(1 for c in text if c in ['.', '!', '?'])

def main():
    text = get_string("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    # Coleman-Liau index
    L = (letters / words) * 100
    S = (sentences / words) * 100
    index = round(0.0588 * L - 0.296 * S - 15.8)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")

if __name__ == "__main__":
    main()
